components/script/dom/webgl2renderingcontext.rs

```diff
@@ -147,6 +156,196 @@ impl WebGL2RenderingContext {
             slot.set(None);
         }
     }
+
+    fn calc_read_pixel_formats(
+        &self,
+        pixel_type: u32,
+        format: u32,
+    ) -> Result<(&[Type], usize), WebGLError> {
```

Comment:
Please introduce a structure type for the tuple, or at least document what its two items are supposed to mean.